File: .agent/skills/cnma-cap-backend/templates/node/srv/src/domain/__feature__/events/OnBeforeCreate{{FeatureName}}Event.js
import cds from '@sap/cds';

/**
 * OnBeforeCreate{{FeatureName}}Event - BEFORE CREATE event handler.
 * SRP: Validates data before creation.
 */
module.exports = {
    execute: async (req) => {
        try {
            const userId = req.user?.id || 'anonymous';
            const data = req.data;

            // Auto-fill UUID if not provided
            if (!data.ID) {
                data.ID = cds.utils.uuid();
            }

            // Set audit fields
            data.createdBy = userId;
            data.modifiedBy = userId;

            // Validation logic here
            if (!data || Object.keys(data).length === 0) {
                req.error(400, 'Invalid data: empty payload');
            }

            // Business validation - check for conflicts
            const conflictQuery = cds
                .select('{{FeatureName}}')
                .where({ createdBy: userId })
                .limit(1);

            const existing = await cds.run(conflictQuery);
            if (existing && existing.length > 0) {
                req.error(409, 'Conflict: record already exists');
            }
        } catch (error) {
            req.error(500, 'Error in before create: ' + error.message);
        }
    }
};

File: .agent/skills/cnma-cap-backend/templates/java/srv/src/main/java/cnma/{{module_name}}/domain/__feature__/events/OnBeforeCreate{{FeatureName}}Event.java
package cnma.{{module_name}}.domain.__feature__.events;

import com.sap.cds.services.RequestContext;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

/**
 * OnBeforeCreate{{FeatureName}}Event - BEFORE CREATE event handler.
 * SRP: Validates data before creation, sets audit fields.
 */
@Slf4j
@Component
@RequiredArgsConstructor
public class OnBeforeCreate{{FeatureName}}Event {

    private final RequestContext requestContext;

    public void execute(Object req) {
        String userId = getUserId(req);
        log.debug("BEFORE CREATE {{FeatureName}} by user: {}", userId);

        // Auto-fill UUID if not provided
        // Set audit fields (createdBy, modifiedBy)
        // Validate data
        // Check for conflicts
    }

    private String getUserId(Object req) {
        return "anonymous";
    }
}